### PR TITLE
Add Without Preparations btn to Gift and Loan

### LINKED ISSUE #4076: `v7.9.1-prerelease`: Cannot create a loan without prepartions
image

**Left:** `v7.9.0`
**RIght:** `v7.9.1-prerelease`

Different databases

https://herbrbge9323-v791-prerelease.test.specifysystems.org/specify/view/loan/new/

Arc browser, macOS Sonoma

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4076

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ Specify 7. It is possible to copy the Specify 6 install from another
 Linux system to avoid the need to install Java on the server.
 
 ```shell
-wget https://update.specifysoftware.org/6802/Specify_unix_64.sh
+wget https://update.specifysoftware.org/Specify_unix_64.sh
```
Comment (maxxxxxdlp) [inline README.md:227]: @CarolineDenis you selected a wrong merge target for this PR
whatever branch you created `issue-4076` from, that is the branch it should be merging in

File: specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx [hunk 2]
```diff
@@ -288,7 +288,12 @@ export function InteractionDialog({
                 >
                   {interactionsText.addUnassociated()}
                 </Button.Info>
-              ) : model.name === 'Loan' || action.model.name === 'Loan' ? (
+              ) : (model.name === 'Loan' || action.model.name === 'Loan') &&
+                !(
+                  state.type === 'MissingState' &&
+                  prepsData?.length !== 0 &&
+                  prepsData
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx:295]: can you double-check your code to see what does `&& prepsData` do in this code snippet under different possible values of `prepsData` to see if it is necessary

File: specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx [hunk 3]
```diff
@@ -288,7 +288,12 @@ export function InteractionDialog({
                 >
                   {interactionsText.addUnassociated()}
                 </Button.Info>
-              ) : model.name === 'Loan' || action.model.name === 'Loan' ? (
+              ) : (model.name === 'Loan' || action.model.name === 'Loan') &&
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx:291]: looks hacky. can you analyze the code/use debugger to see what are the possible values of `model.name` and `action.model.name` and what are the relations between them - chances are you only need one of these 2 conditions

Comment (maxxxxxdlp) [reply specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx:291]: (though to be fair to you, the very existence of both `model.name` and `action.model.name` in this file is a hack I wasnt' able to get rid of when refactoring this file)

File: specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx [hunk 4]
```diff
@@ -288,12 +288,22 @@ export function InteractionDialog({
...
+                  prepsData?.length !== 0 &&
+                  prepsData
+                ) ? (
                 <Link.Blue href={getResourceViewUrl('Loan')}>
                   {interactionsText.withoutPreparations()}
                 </Link.Blue>
-              ) : undefined}
+              ) : null}
+              {model.name === 'Gift' ||
+              (action.model.name === 'Gift' && itemCollection === undefined) ? (
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx:302]: sorry I didn't read the surrounding code much, but it looks suspicious that the condition for Loan button and Gift button look so different

Also, can you please look at the code in version `v7.9.0` or before to see how this looked like before the #4076 bug?
If the task is to bring back a button that was accidentally removed, then the code that was here before the change would be a good guide

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]

File: specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx (hunk 3 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx:291]: using only action.model.name is working

File: specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx (hunk 4 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx:302]: it didn't exist before. We only had a button for loans

**Comment by CarolineDenis** [comment]
NOTES: Created a new branch with correct base. 
https://github.com/specify/specify7/pull/4093

**Pull request closed without merging**